### ISSUE TITLE
Add revision links to JSON API payloads

### ISSUE THREAD

**Issue by e0ipso** [Lullabot/ibm-dte#4]
## Problem

We need to allow consumer applications to discover and interact with revisions using hypermedia. We also need this solution to not be Drupal specific.

## Proposed resolution

Implement [RFC 5829](https://tools.ietf.org/html/rfc5829) in JSON API. For now we are only interested in `latest-version` and `working-copy`. We can postpone or even descope the rest of link types.

This will require decorating the JSON API Link Manager service to add additional (non-JSON API specific) links.

## Notes

This ticket depends on #3 for the actual links.

**Comment by e0ipso** [comment]
Progress against this is being tracked in https://www.drupal.org/project/jsonapi/issues/2992836.

**Comment by Snugug** [comment]
I've got the latest versions working in my repo!
